### DISCUSSION TITLE
[BUG] NameError: undefined local variable or method `recipient' for NewMessageNotifier:Class

### DISCUSSION THREAD

**Discussion by mnifakram** [excid3/noticed#505] [Q&A] [answered]
## Bug Report

**Describe the Bug:**
NameError: undefined local variable or method 'recipient' for NewMessageNotifier:Class
Did you mean?  recipients
               _recipients
               _recipients?
               _recipients=
from /Users/akram/.rbenv/versions/3.0.3/lib/ruby/gems/3.0.0/gems/activerecord-7.0.4.2/lib/active_record/dynamic_matchers.rb:22:in 'method_missing'

**To Reproduce:**
**Notifier class:**

```
class NewMessageNotifier < ApplicationNotifier
  deliver_by :action_cable do |config|
    config.channel = 'DashboardNotificationsChannel'
    config.stream = "dashboard_notifications_#{recipient.type}_#{recipient.id}"
    config.message = params.to_json
  end
end
```

**Call statement**

`NewMessageNotifier.with(record: record, params: { snapshot: record.content }).deliver(recipients)`

If I comment the config.stream line, it does create the event and all notifications with the correct recipients. 

Not sure if it's the same as #366  

**Environment:**
- Noticed gem version: 2.4
- Ruby version: 3.0.3
- Rails version: 7.0.4.1
- Operating System: Ventura 13.5.1

**Comment by excid3** [comment, marked as answer]
You can't use the `recipient` variable at the class level. You would use it inside a lambda that gets evaluated later on when it's being delivered.

**Reply by mnifakram** [reply]
Thank you so much for your help
